tests/clang_format_utils.py

```diff
@@ -0,0 +1,78 @@
+# Copyright 2020 MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# this file is adapted from
+# github/pytorch/pytorch/blob/63d62d3e44a0a4ec09d94f30381d49b78cc5b095/tools/clang_format_utils.py
+
+import os
+import platform
+import stat
+import sys
+
+from monai.apps.utils import download_url
+
+# String representing the host platform (e.g. Linux, Darwin).
+HOST_PLATFORM = platform.system()
+
+# MONAI directory root, derived from the location of this file.
+MONAI_ROOT = os.path.dirname(os.path.dirname(os.path.realpath(__file__)))
+
+# This dictionary maps each platform to the S3 object URL for its clang-format binary.
+PLATFORM_TO_CF_URL = {
+    "Darwin": "https://oss-clang-format.s3.us-east-2.amazonaws.com/mac/clang-format-mojave",
+    "Linux": "https://oss-clang-format.s3.us-east-2.amazonaws.com/linux64/clang-format-linux64",
+}
+
+# This dictionary maps each platform to a relative path to a file containing its reference hash.
+# github/pytorch/pytorch/tree/63d62d3e44a0a4ec09d94f30381d49b78cc5b095/tools/clang_format_hash
+PLATFORM_TO_HASH = {
+    "Darwin": "020c7f38f14665c2ed82f3e8976c9074c2cfac0a",
+    "Linux": "d1365110da598d148d8143a7f2ccfd8bac7df499",
+}
+
+# Directory and file paths for the clang-format binary.
+CLANG_FORMAT_DIR = os.path.join(MONAI_ROOT, ".clang-format-bin")
+CLANG_FORMAT_PATH = os.path.join(CLANG_FORMAT_DIR, "clang-format")
+
+
+def get_and_check_clang_format():
```

Comment:
clang-format can be had through pip instead: `pip install clang-format`. I think downloading binaries from remote sources, even if we're checking against a hash, isn't a secure practice.

Reply:
pip only provides linux versions of clang-format.  Mac is not supported, and windows is not supported.